internal/repositories/accountMongoRepository.go

```
@@ -0,0 +1,149 @@
+
+	if err != nil {
+		logging.Fatal(err)
+		return nil, errors.New("database error")
+	}
+
+	defer results.Close(ctx)
+	for results.Next(ctx) {
+		var account models.Account
+		if err = results.Decode(&account); err != nil {
+			logging.Fatal(err)
+			return nil, errors.New("decode error")
+		}
+
+		accounts = append(accounts, account)
+	}
+	logging.Infof("find all accounts")
+	return accounts, err
+}
+
+func (r repository) FindOne(id string) (models.Account, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	var user models.Account
+	defer cancel()
+
+	objId, _ := primitive.ObjectIDFromHex(id)
+
+	err := accountsCollections.FindOne(ctx, bson.M{"id": objId}).Decode(&user)
+
+	if err != nil {
+		logging.Error(err)
+		return models.Account{}, errors.New("database error")
+	}
+
+	logging.Infof("Find account by id, email: ", user.Email)
+	return user, err
+}
+
+func (r repository) Update(id string, user models.Account) (models.Account, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	objId, _ := primitive.ObjectIDFromHex(id)
+
+	update := bson.M{"email": user.Email, "password": user.Password, "username": user.Username}
+
+	result, err := accountsCollections.UpdateOne(ctx, bson.M{"id": objId}, bson.M{"$set": update})
+
+	if err != nil {
+		logging.Error(err)
+		return models.Account{}, errors.New("database error")
+	}
+
+	var updatedUser models.Account
+	if result.MatchedCount == 1 {
+		err := accountsCollections.FindOne(ctx, bson.M{"id": objId}).Decode(&updatedUser)
+		if err != nil {
+			logging.Error(err)
+			return models.Account{}, errors.New("database error")
+		}
+	}
+
+	logging.Infof("Update account", user.Email)
+	return updatedUser, err
+}
+
+func (r repository) Delete(id string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	objId, _ := primitive.ObjectIDFromHex(id)
```

You should not ignore an error here. It is best practice always to check it